### PR TITLE
add change visibility blog post perm

### PR DESCRIPTION
**Pull request by Riolku**
This allows us to have two groups: users who can draft blog posts, that only admins can public, and of course, admins. It works much the same as problems and contests do.

### CODE REVIEW

**Comment by Riolku** [comment]
Prefix perm checks with `judge.`, the code is wrong otherwise. In fact, it silently returns false.

**Comment by Riolku** [comment]
Use a list instead of a tuple in `get_readonly_fields` for better consistency with the rest of the codebase.

File: judge/admin/interface.py [hunk 1]
```diff
@@ -79,11 +79,27 @@ class BlogPostAdmin(VersionAdmin):
     form = BlogPostForm
     date_hierarchy = 'publish_on'
 
+    def has_view_permission(self, request, obj=None):
```
Comment (Ninjaclasher) [inline judge/admin/interface.py:82]: Is this necessary? None of the other classes override `has_view_permission`.

File: judge/admin/interface.py [hunk 2]
```diff
@@ -79,11 +79,27 @@ class BlogPostAdmin(VersionAdmin):
...
 
+    def get_readonly_fields(self, request, obj=None):
+        if not request.user.has_perm('judge.change_post_visibility'):
+            return ['visible']
+        return []
+
+    def get_queryset(self, request):
+        queryset = BlogPost.objects.all()
+        if not request.user.has_perm('judge.edit_all_post'):
+            queryset = queryset.filter(authors=request.user.profile)
```
Comment (Ninjaclasher) [inline judge/admin/interface.py:100]:
```suggestion
            queryset = queryset.filter(authors=request.profile)
```

**Review by Ninjaclasher — COMMENTED** [review]
Otherwise LGTM

File: judge/admin/interface.py (hunk 1 above)
Comment (Riolku) [reply judge/admin/interface.py:82]: Might as well document this: after a long search, it seems that `get_queryset` is filtered to fetch the model instance, so as long as we write that method write, we don't have to write `has_view_permission`.

File: judge/admin/interface.py (hunk 2 above)
Comment (Riolku) [reply judge/admin/interface.py:100]: Did not know request had a profile field, thanks.

**Comment by Riolku** [comment]
Alright, updated, thanks.

**Comment by Riolku** [comment]
(forgot to push)

**Comment by Riolku** [comment]
Rename migration name to something more useful.

**Review by kiritofeng — APPROVED** [review]
Please rebase migration, probably

**Pull request merged**